Operators_and_constructions.rb

```diff
@@ -11,41 +11,46 @@ def gets_random_10
 
   count = 0
 
-while count < 10
-  num = rand(9)
- first_string = "=========" + " #{num}"
-
-  if num.odd?  and num!=0
-    result = first_string + " the number is odd"
-     if num < 5
-       puts result + " the number is less than 5"
-       elsif num > 5
-         puts result + " the number is more than 5"
-        else
-       puts result + " the number is 5"
+  while count < 10
+    @num = rand(9)
+    divider = "========="
+
+    case
+    when @num.odd? && @num!=0
+      @result = "#{@num}" + " the number is odd"
+      puts divider
+      check_if_five_exists
+    when @num.even? && @num!=0
+      puts divider
+      @result = "#{@num}" + " the number is even"
+      check_if_five_exists
+    when @num == 0
```

Comment:
If you place this block ( when @num == 0) at the first place inside case statement - you won't need to add to other conditions && @num!=0

Other is OK